lua/cfc_time/server/time/sv_mysql.lua

```diff
@@ -0,0 +1,265 @@
+end
+
+function storage:PrepareStatements()
+    logger:info( "Constructing prepared statements..." )
+
+    local realm = self.realm
+
+    local newUser = "INSERT IGNORE INTO users (steam_id) VALUES(?)"
+
+    local newSession = string.format( [[
+        INSERT INTO sessions (user_id, joined, departed, duration, realm) VALUES(?, ?, ?, ?, '%s')
+    ]], realm )
+
+    local totalTime = string.format( [[
+        SELECT SUM(duration)
+        FROM sessions
+        WHERE user_id = ?
+        AND realm = '%s'
+    ]], realm )
+
+    local latestSession = string.format( [[
+        SELECT *
+        FROM sessions
+        WHERE user_id = ?
+        AND realm = '%s'
+        ORDER BY joined DESC
+        LIMIT 1
+    ]], realm )
+
+    self:AddPreparedStatement( "newUser", newUser )
+    self:AddPreparedStatement( "newSession", newSession )
+    self:AddPreparedStatement( "totalTime", totalTime )
+    self:AddPreparedStatement( "latestSession", latestSession )
+end
+
+function storage:Prepare( statementName, onSuccess, ... )
+    local query = self.preparedQueries[statementName]
+    query:clearParameters()
+
+    for k, v in pairs( { ... } ) do
+        if isnumber( v ) then
+            query:setNumber( k, v )
+        elseif isstring( v ) then
+            query:setString( k, v )
+        elseif isbool( v ) then
+            query:setBoolean( k, v )
+        elseif v == nil then
+            query:setNull( k )
+        else
+            error( "Wrong data type passed to Prepare statement!: " .. v )
+        end
+    end
+
+    if onSuccess then query.onSuccess = onSuccess end
+
+    return query
+end
+
+function storage.database:onConnected()
+    logger:info( "DB successfully connected! Beginning init..." )
+
+    local transaction = storage:InitTransaction()
+
+    transaction:addQuery( storage:CreateUsersQuery() )
+    transaction:addQuery( storage:CreateSessionsQuery() )
+    transaction:addQuery( storage:SessionCleanupQuery() )
+
+    transaction.onSuccess = function()
+        storage:PrepareStatements()
+    end
+
+    transaction:start()
+end
+
+function storage.database:onConnectionFailed( err )
+    logger:error( "Failed to connect to database!" )
+    logger:fatal( err )
+end
+
+hook.Add( "PostGamemodeLoaded", "CFC_Time_DBInit", function()
+    logger:info( "Gamemoded loaded, beginning database init..." )
+    storage.database:connect()
+end )
+
+-- TODO: Find a better/safer way to do this
+function storage:BuildSessionUpdate( data, id )
+    local updateSection = "UPDATE sessions "
+    local setSection = "SET "
+    local whereSection = string.format(
+        "WHERE id = %s AND realm = '%s'",
+        id, self.realm
+    )
+
+    -- TODO: Have a safeguard here for invalid keys?
+    local count = table.Count( data )
+    local idx = 1
+
+    -- FIXME: This will break because we pass strings/ints interchangeably, but strings need quotes around them
+    for k, v in pairs( data ) do
+        local newSet = k .. " = " .. v
```

Comment:
```suggestion
        if type( v ) == "string" then
            v = "\"" .. v .. "\""
        end
        
        local newSet = k .. " = " .. v
```

Something like this?